app/models/miq_server/worker_management/kubernetes.rb

```diff
@@ -24,7 +24,9 @@ def sync_starting_workers
     MiqWorker.find_all_starting.each do |worker|
       next if worker.class.rails_worker?
 
-      worker_pod = get_pod(worker[:system_uid])
+      worker_pod = current_pods[worker[:system_uid]]
```

Comment:
This is the only change I need to think about... The logic looks right.  I'm not sure if this will have side effects when the cache is not yet populated with the current pods.  Previously, we relied on a side effect.  I'm don't remember why it bypassed the collect_initial/watch_for_events "current_pods" cache and went straight to get_pods.

Reply:
Yeah I'm assuming it was because that the cache wasn't guaranteed to be populated by the time we're here which is what waiting for the initial collection to be done before returning